src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1366,6 +1415,14 @@ private void onDecodeConnect(
                 keepAlive = packet.keepAlive();
                 keepAliveTimeout = Math.round(keepAlive * 1.5 * 1000);
                 doSignalKeepAliveTimeoutIfNecessary();
+                if (sessionsAvailable)
+                {
+                    resolveSession(traceId, authorization, packet.flags(),
```

Comment:
Let's put a block around the `for` loop above and move the label to that block.
```
byte reasonCode = SUCCESS;
decode:
{
    if (connected)
    {
        reasonCode = PROTOCOL_ERROR;
        break decode;
    }
    ...
    for ( ...
}
```
Inside the block, we can `break decode` after setting the `reasonCode` to something other than `SUCCESS`.
After the block, we can have a condition handling non-success.
```
if (reasonCode != SUCCESS)
{
    doNetworkEnd(traceId, authorization);
}
```
This approach will allow local variables such as `mqttConnectPayload` to remain in scope inside the block, avoiding the need to use global variables such as `mqttConnectPayloadRO` instead.